// @aws-sdk/client-health/dist-cjs/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  ConcurrentModificationException: () => ConcurrentModificationException,
  DescribeAffectedAccountsForOrganizationCommand: () => DescribeAffectedAccountsForOrganizationCommand,
  DescribeAffectedEntitiesCommand: () => DescribeAffectedEntitiesCommand,
  DescribeAffectedEntitiesForOrganizationCommand: () => DescribeAffectedEntitiesForOrganizationCommand,
  DescribeEntityAggregatesCommand: () => DescribeEntityAggregatesCommand,
  DescribeEntityAggregatesForOrganizationCommand: () => DescribeEntityAggregatesForOrganizationCommand,
  DescribeEventAggregatesCommand: () => DescribeEventAggregatesCommand,
  DescribeEventDetailsCommand: () => DescribeEventDetailsCommand,
  DescribeEventDetailsForOrganizationCommand: () => DescribeEventDetailsForOrganizationCommand,
  DescribeEventTypesCommand: () => DescribeEventTypesCommand,
  DescribeEventsCommand: () => DescribeEventsCommand,
  DescribeEventsForOrganizationCommand: () => DescribeEventsForOrganizationCommand,
  DescribeHealthServiceStatusForOrganizationCommand: () => DescribeHealthServiceStatusForOrganizationCommand,
  DisableHealthServiceAccessForOrganizationCommand: () => DisableHealthServiceAccessForOrganizationCommand,
  EnableHealthServiceAccessForOrganizationCommand: () => EnableHealthServiceAccessForOrganizationCommand,
  EntityStatusCode: () => EntityStatusCode,
  EventAggregateField: () => EventAggregateField,
  EventScopeCode: () => EventScopeCode,
  EventStatusCode: () => EventStatusCode,
  EventTypeCategory: () => EventTypeCategory,
  Health: () => Health,
  HealthClient: () => HealthClient,
  HealthServiceException: () => HealthServiceException,
  InvalidPaginationToken: () => InvalidPaginationToken,
  UnsupportedLocale: () => UnsupportedLocale,
  __Client: () => import_smithy_client.Client,
  paginateDescribeAffectedAccountsForOrganization: () => paginateDescribeAffectedAccountsForOrganization,
  paginateDescribeAffectedEntities: () => paginateDescribeAffectedEntities,
  paginateDescribeAffectedEntitiesForOrganization: () => paginateDescribeAffectedEntitiesForOrganization,
  paginateDescribeEventAggregates: () => paginateDescribeEventAggregates,
  paginateDescribeEventTypes: () => paginateDescribeEventTypes,
  paginateDescribeEvents: () => paginateDescribeEvents,
  paginateDescribeEventsForOrganization: () => paginateDescribeEventsForOrganization
});
module.exports = __toCommonJS(index_exports);

// src/HealthClient.ts
var import_middleware_host_header = require("@aws-sdk/middleware-host-header");
var import_middleware_logger = require("@aws-sdk/middleware-logger");
var import_middleware_recursion_detection = require("@aws-sdk/middleware-recursion-detection");
var import_middleware_user_agent = require("@aws-sdk/middleware-user-agent");
var import_config_resolver = require("@smithy/config-resolver");
var import_core = require("@smithy/core");
var import_middleware_content_length = require("@smithy/middleware-content-length");
var import_middleware_endpoint = require("@smithy/middleware-endpoint");
var import_middleware_retry = require("@smithy/middleware-retry");

var import_httpAuthSchemeProvider = require("./auth/httpAuthSchemeProvider");

// src/endpoint/EndpointParameters.ts
var resolveClientEndpointParameters = /* @__PURE__ */ __name((options) => {
  return Object.assign(options, {
    useDualstackEndpoint: options.useDualstackEndpoint ?? false,
    useFipsEndpoint: options.useFipsEndpoint ?? false,
    defaultSigningName: "health"
  });
}, "resolveClientEndpointParameters");
var commonParams = {
  UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
  Endpoint: { type: "builtInParams", name: "endpoint" },
  Region: { type: "builtInParams", name: "region" },
  UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" }
};

// src/HealthClient.ts
var import_runtimeConfig = require("././runtimeConfig");

// src/runtimeExtensions.ts
var import_region_config_resolver = require("@aws-sdk/region-config-resolver");
var import_protocol_http = require("@smithy/protocol-http");
var import_smithy_client = require("@smithy/smithy-client");

// src/auth/httpAuthExtensionConfiguration.ts
var getHttpAuthExtensionConfiguration = /* @__PURE__ */ __name((runtimeConfig) => {
  const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
  let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
  let _credentials = runtimeConfig.credentials;
  return {
    setHttpAuthScheme(httpAuthScheme) {
      const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
      if (index === -1) {
        _httpAuthSchemes.push(httpAuthScheme);
      } else {
        _httpAuthSchemes.splice(index, 1, httpAuthScheme);
      }
    },
    httpAuthSchemes() {
      return _httpAuthSchemes;
    },
    setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
      _httpAuthSchemeProvider = httpAuthSchemeProvider;
    },
    httpAuthSchemeProvider() {
      return _httpAuthSchemeProvider;
    },
    setCredentials(credentials) {
      _credentials = credentials;
    },
    credentials() {
      return _credentials;
    }
  };
}, "getHttpAuthExtensionConfiguration");
var resolveHttpAuthRuntimeConfig = /* @__PURE__ */ __name((config) => {
  return {
    httpAuthSchemes: config.httpAuthSchemes(),
    httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
    credentials: config.credentials()
  };
}, "resolveHttpAuthRuntimeConfig");

// src/runtimeExtensions.ts
var resolveRuntimeExtensions = /* @__PURE__ */ __name((runtimeConfig, extensions) => {
  const extensionConfiguration = Object.assign(
    (0, import_region_config_resolver.getAwsRegionExtensionConfiguration)(runtimeConfig),
    (0, import_smithy_client.getDefaultExtensionConfiguration)(runtimeConfig),
    (0, import_protocol_http.getHttpHandlerExtensionConfiguration)(runtimeConfig),
    getHttpAuthExtensionConfiguration(runtimeConfig)
  );
  extensions.forEach((extension) => extension.configure(extensionConfiguration));
  return Object.assign(
    runtimeConfig,
    (0, import_region_config_resolver.resolveAwsRegionExtensionConfiguration)(extensionConfiguration),
    (0, import_smithy_client.resolveDefaultRuntimeConfig)(extensionConfiguration),
    (0, import_protocol_http.resolveHttpHandlerRuntimeConfig)(extensionConfiguration),
    resolveHttpAuthRuntimeConfig(extensionConfiguration)
  );
}, "resolveRuntimeExtensions");

// src/HealthClient.ts
var HealthClient = class extends import_smithy_client.Client {
  static {
    __name(this, "HealthClient");
  }
  /**
   * The resolved configuration of HealthClient class. This is resolved and normalized from the {@link HealthClientConfig | constructor configuration interface}.
   */
  config;
  constructor(...[configuration]) {
    const _config_0 = (0, import_runtimeConfig.getRuntimeConfig)(configuration || {});
    super(_config_0);
    this.initConfig = _config_0;
    const _config_1 = resolveClientEndpointParameters(_config_0);
    const _config_2 = (0, import_middleware_user_agent.resolveUserAgentConfig)(_config_1);
    const _config_3 = (0, import_middleware_retry.resolveRetryConfig)(_config_2);
    const _config_4 = (0, import_config_resolver.resolveRegionConfig)(_config_3);
    const _config_5 = (0, import_middleware_host_header.resolveHostHeaderConfig)(_config_4);
    const _config_6 = (0, import_middleware_endpoint.resolveEndpointConfig)(_config_5);
    const _config_7 = (0, import_httpAuthSchemeProvider.resolveHttpAuthSchemeConfig)(_config_6);
    const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
    this.config = _config_8;
    this.middlewareStack.use((0, import_middleware_user_agent.getUserAgentPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_retry.getRetryPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_content_length.getContentLengthPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_host_header.getHostHeaderPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_logger.getLoggerPlugin)(this.config));
    this.middlewareStack.use((0, import_middleware_recursion_detection.getRecursionDetectionPlugin)(this.config));
    this.middlewareStack.use(
      (0, import_core.getHttpAuthSchemeEndpointRuleSetPlugin)(this.config, {
        httpAuthSchemeParametersProvider: import_httpAuthSchemeProvider.defaultHealthHttpAuthSchemeParametersProvider,
        identityProviderConfigProvider: /* @__PURE__ */ __name(async (config) => new import_core.DefaultIdentityProviderConfig({
          "aws.auth#sigv4": config.credentials
        }), "identityProviderConfigProvider")
      })
    );
    this.middlewareStack.use((0, import_core.getHttpSigningPlugin)(this.config));
  }
  /**
   * Destroy underlying resources, like sockets. It's usually not necessary to do this.
   * However in Node.js, it's best to explicitly shut down the client's agent when it is no longer needed.
   * Otherwise, sockets might stay open for quite a long time before the server terminates them.
   */
  destroy() {
    super.destroy();
  }
};

// src/Health.ts


// src/commands/DescribeAffectedAccountsForOrganizationCommand.ts

var import_middleware_serde = require("@smithy/middleware-serde");


// src/protocols/Aws_json1_1.ts
var import_core2 = require("@aws-sdk/core");



// src/models/HealthServiceException.ts

var HealthServiceException = class _HealthServiceException extends import_smithy_client.ServiceException {
  static {
    __name(this, "HealthServiceException");
  }
  /**
   * @internal
   */
  constructor(options) {
    super(options);
    Object.setPrototypeOf(this, _HealthServiceException.prototype);
  }
};

// src/models/models_0.ts
var EntityStatusCode = {
  IMPAIRED: "IMPAIRED",
  PENDING: "PENDING",
  RESOLVED: "RESOLVED",
  UNIMPAIRED: "UNIMPAIRED",
  UNKNOWN: "UNKNOWN"
};
var EventScopeCode = {
  ACCOUNT_SPECIFIC: "ACCOUNT_SPECIFIC",
  NONE: "NONE",
  PUBLIC: "PUBLIC"
};
var InvalidPaginationToken = class _InvalidPaginationToken extends HealthServiceException {
  static {
    __name(this, "InvalidPaginationToken");
  }
  name = "InvalidPaginationToken";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "InvalidPaginationToken",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _InvalidPaginationToken.prototype);
  }
};
var UnsupportedLocale = class _UnsupportedLocale extends HealthServiceException {
  static {
    __name(this, "UnsupportedLocale");
  }
  name = "UnsupportedLocale";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "UnsupportedLocale",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _UnsupportedLocale.prototype);
  }
};
var EventAggregateField = {
  EventTypeCategory: "eventTypeCategory"
};
var EventStatusCode = {
  CLOSED: "closed",
  OPEN: "open",
  UPCOMING: "upcoming"
};
var EventTypeCategory = {
  ACCOUNT_NOTIFICATION: "accountNotification",
  INVESTIGATION: "investigation",
  ISSUE: "issue",
  SCHEDULED_CHANGE: "scheduledChange"
};
var ConcurrentModificationException = class _ConcurrentModificationException extends HealthServiceException {
  static {
    __name(this, "ConcurrentModificationException");
  }
  name = "ConcurrentModificationException";
  $fault = "client";
  /**
   * @internal
   */
  constructor(opts) {
    super({
      name: "ConcurrentModificationException",
      $fault: "client",
      ...opts
    });
    Object.setPrototypeOf(this, _ConcurrentModificationException.prototype);
  }
};

// src/protocols/Aws_json1_1.ts
var se_DescribeAffectedAccountsForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAffectedAccountsForOrganization");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAffectedAccountsForOrganizationCommand");
var se_DescribeAffectedEntitiesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAffectedEntities");
  let body;
  body = JSON.stringify(se_DescribeAffectedEntitiesRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAffectedEntitiesCommand");
var se_DescribeAffectedEntitiesForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeAffectedEntitiesForOrganization");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeAffectedEntitiesForOrganizationCommand");
var se_DescribeEntityAggregatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEntityAggregates");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEntityAggregatesCommand");
var se_DescribeEntityAggregatesForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEntityAggregatesForOrganization");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEntityAggregatesForOrganizationCommand");
var se_DescribeEventAggregatesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEventAggregates");
  let body;
  body = JSON.stringify(se_DescribeEventAggregatesRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventAggregatesCommand");
var se_DescribeEventDetailsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEventDetails");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventDetailsCommand");
var se_DescribeEventDetailsForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEventDetailsForOrganization");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventDetailsForOrganizationCommand");
var se_DescribeEventsCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEvents");
  let body;
  body = JSON.stringify(se_DescribeEventsRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventsCommand");
var se_DescribeEventsForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEventsForOrganization");
  let body;
  body = JSON.stringify(se_DescribeEventsForOrganizationRequest(input, context));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventsForOrganizationCommand");
var se_DescribeEventTypesCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeEventTypes");
  let body;
  body = JSON.stringify((0, import_smithy_client._json)(input));
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeEventTypesCommand");
var se_DescribeHealthServiceStatusForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DescribeHealthServiceStatusForOrganization");
  const body = "{}";
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DescribeHealthServiceStatusForOrganizationCommand");
var se_DisableHealthServiceAccessForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("DisableHealthServiceAccessForOrganization");
  const body = "{}";
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_DisableHealthServiceAccessForOrganizationCommand");
var se_EnableHealthServiceAccessForOrganizationCommand = /* @__PURE__ */ __name(async (input, context) => {
  const headers = sharedHeaders("EnableHealthServiceAccessForOrganization");
  const body = "{}";
  return buildHttpRpcRequest(context, headers, "/", void 0, body);
}, "se_EnableHealthServiceAccessForOrganizationCommand");
var de_DescribeAffectedAccountsForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAffectedAccountsForOrganizationCommand");
var de_DescribeAffectedEntitiesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAffectedEntitiesResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAffectedEntitiesCommand");
var de_DescribeAffectedEntitiesForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeAffectedEntitiesForOrganizationResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeAffectedEntitiesForOrganizationCommand");
var de_DescribeEntityAggregatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEntityAggregatesCommand");
var de_DescribeEntityAggregatesForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEntityAggregatesForOrganizationCommand");
var de_DescribeEventAggregatesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventAggregatesCommand");
var de_DescribeEventDetailsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeEventDetailsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventDetailsCommand");
var de_DescribeEventDetailsForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeEventDetailsForOrganizationResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventDetailsForOrganizationCommand");
var de_DescribeEventsCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeEventsResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventsCommand");
var de_DescribeEventsForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = de_DescribeEventsForOrganizationResponse(data, context);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventsForOrganizationCommand");
var de_DescribeEventTypesCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeEventTypesCommand");
var de_DescribeHealthServiceStatusForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  const data = await (0, import_core2.parseJsonBody)(output.body, context);
  let contents = {};
  contents = (0, import_smithy_client._json)(data);
  const response = {
    $metadata: deserializeMetadata(output),
    ...contents
  };
  return response;
}, "de_DescribeHealthServiceStatusForOrganizationCommand");
var de_DisableHealthServiceAccessForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_DisableHealthServiceAccessForOrganizationCommand");
var de_EnableHealthServiceAccessForOrganizationCommand = /* @__PURE__ */ __name(async (output, context) => {
  if (output.statusCode >= 300) {
    return de_CommandError(output, context);
  }
  await (0, import_smithy_client.collectBody)(output.body, context);
  const response = {
    $metadata: deserializeMetadata(output)
  };
  return response;
}, "de_EnableHealthServiceAccessForOrganizationCommand");
var de_CommandError = /* @__PURE__ */ __name(async (output, context) => {
  const parsedOutput = {
    ...output,
    body: await (0, import_core2.parseJsonErrorBody)(output.body, context)
  };
  const errorCode = (0, import_core2.loadRestJsonErrorCode)(output, parsedOutput.body);
  switch (errorCode) {
    case "InvalidPaginationToken":
    case "com.amazonaws.health#InvalidPaginationToken":
      throw await de_InvalidPaginationTokenRes(parsedOutput, context);
    case "UnsupportedLocale":
    case "com.amazonaws.health#UnsupportedLocale":
      throw await de_UnsupportedLocaleRes(parsedOutput, context);
    case "ConcurrentModificationException":
    case "com.amazonaws.health#ConcurrentModificationException":
      throw await de_ConcurrentModificationExceptionRes(parsedOutput, context);
    default:
      const parsedBody = parsedOutput.body;
      return throwDefaultError({
        output,
        parsedBody,
        errorCode
      });
  }
}, "de_CommandError");
var de_ConcurrentModificationExceptionRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new ConcurrentModificationException({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_ConcurrentModificationExceptionRes");
var de_InvalidPaginationTokenRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new InvalidPaginationToken({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_InvalidPaginationTokenRes");
var de_UnsupportedLocaleRes = /* @__PURE__ */ __name(async (parsedOutput, context) => {
  const body = parsedOutput.body;
  const deserialized = (0, import_smithy_client._json)(body);
  const exception = new UnsupportedLocale({
    $metadata: deserializeMetadata(parsedOutput),
    ...deserialized
  });
  return (0, import_smithy_client.decorateServiceException)(exception, body);
}, "de_UnsupportedLocaleRes");
var se_DateTimeRange = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    from: /* @__PURE__ */ __name((_) => _.getTime() / 1e3, "from"),
    to: /* @__PURE__ */ __name((_) => _.getTime() / 1e3, "to")
  });
}, "se_DateTimeRange");
var se_dateTimeRangeList = /* @__PURE__ */ __name((input, context) => {
  return input.filter((e) => e != null).map((entry) => {
    return se_DateTimeRange(entry, context);
  });
}, "se_dateTimeRangeList");
var se_DescribeAffectedEntitiesRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    filter: /* @__PURE__ */ __name((_) => se_EntityFilter(_, context), "filter"),
    locale: [],
    maxResults: [],
    nextToken: []
  });
}, "se_DescribeAffectedEntitiesRequest");
var se_DescribeEventAggregatesRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    aggregateField: [],
    filter: /* @__PURE__ */ __name((_) => se_EventFilter(_, context), "filter"),
    maxResults: [],
    nextToken: []
  });
}, "se_DescribeEventAggregatesRequest");
var se_DescribeEventsForOrganizationRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    filter: /* @__PURE__ */ __name((_) => se_OrganizationEventFilter(_, context), "filter"),
    locale: [],
    maxResults: [],
    nextToken: []
  });
}, "se_DescribeEventsForOrganizationRequest");
var se_DescribeEventsRequest = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    filter: /* @__PURE__ */ __name((_) => se_EventFilter(_, context), "filter"),
    locale: [],
    maxResults: [],
    nextToken: []
  });
}, "se_DescribeEventsRequest");
var se_EntityFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    entityArns: import_smithy_client._json,
    entityValues: import_smithy_client._json,
    eventArns: import_smithy_client._json,
    lastUpdatedTimes: /* @__PURE__ */ __name((_) => se_dateTimeRangeList(_, context), "lastUpdatedTimes"),
    statusCodes: import_smithy_client._json,
    tags: import_smithy_client._json
  });
}, "se_EntityFilter");
var se_EventFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    availabilityZones: import_smithy_client._json,
    endTimes: /* @__PURE__ */ __name((_) => se_dateTimeRangeList(_, context), "endTimes"),
    entityArns: import_smithy_client._json,
    entityValues: import_smithy_client._json,
    eventArns: import_smithy_client._json,
    eventStatusCodes: import_smithy_client._json,
    eventTypeCategories: import_smithy_client._json,
    eventTypeCodes: import_smithy_client._json,
    lastUpdatedTimes: /* @__PURE__ */ __name((_) => se_dateTimeRangeList(_, context), "lastUpdatedTimes"),
    regions: import_smithy_client._json,
    services: import_smithy_client._json,
    startTimes: /* @__PURE__ */ __name((_) => se_dateTimeRangeList(_, context), "startTimes"),
    tags: import_smithy_client._json
  });
}, "se_EventFilter");
var se_OrganizationEventFilter = /* @__PURE__ */ __name((input, context) => {
  return (0, import_smithy_client.take)(input, {
    awsAccountIds: import_smithy_client._json,
    endTime: /* @__PURE__ */ __name((_) => se_DateTimeRange(_, context), "endTime"),
    entityArns: import_smithy_client._json,
    entityValues: import_smithy_client._json,
    eventStatusCodes: import_smithy_client._json,
    eventTypeCategories: import_smithy_client._json,
    eventTypeCodes: import_smithy_client._json,
    lastUpdatedTime: /* @__PURE__ */ __name((_) => se_DateTimeRange(_, context), "lastUpdatedTime"),
    regions: import_smithy_client._json,
    services: import_smithy_client._json,
    startTime: /* @__PURE__ */ __name((_) => se_DateTimeRange(_, context), "startTime")
  });
}, "se_OrganizationEventFilter");
var de_AffectedEntity = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    awsAccountId: import_smithy_client.expectString,
    entityArn: import_smithy_client.expectString,
    entityMetadata: import_smithy_client._json,
    entityUrl: import_smithy_client.expectString,
    entityValue: import_smithy_client.expectString,
    eventArn: import_smithy_client.expectString,
    lastUpdatedTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "lastUpdatedTime"),
    statusCode: import_smithy_client.expectString,
    tags: import_smithy_client._json
  });
}, "de_AffectedEntity");
var de_DescribeAffectedEntitiesForOrganizationResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    entities: /* @__PURE__ */ __name((_) => de_EntityList(_, context), "entities"),
    failedSet: import_smithy_client._json,
    nextToken: import_smithy_client.expectString
  });
}, "de_DescribeAffectedEntitiesForOrganizationResponse");
var de_DescribeAffectedEntitiesResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    entities: /* @__PURE__ */ __name((_) => de_EntityList(_, context), "entities"),
    nextToken: import_smithy_client.expectString
  });
}, "de_DescribeAffectedEntitiesResponse");
var de_DescribeEventDetailsForOrganizationResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    failedSet: import_smithy_client._json,
    successfulSet: /* @__PURE__ */ __name((_) => de_DescribeEventDetailsForOrganizationSuccessfulSet(_, context), "successfulSet")
  });
}, "de_DescribeEventDetailsForOrganizationResponse");
var de_DescribeEventDetailsForOrganizationSuccessfulSet = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_OrganizationEventDetails(entry, context);
  });
  return retVal;
}, "de_DescribeEventDetailsForOrganizationSuccessfulSet");
var de_DescribeEventDetailsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    failedSet: import_smithy_client._json,
    successfulSet: /* @__PURE__ */ __name((_) => de_DescribeEventDetailsSuccessfulSet(_, context), "successfulSet")
  });
}, "de_DescribeEventDetailsResponse");
var de_DescribeEventDetailsSuccessfulSet = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_EventDetails(entry, context);
  });
  return retVal;
}, "de_DescribeEventDetailsSuccessfulSet");
var de_DescribeEventsForOrganizationResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    events: /* @__PURE__ */ __name((_) => de_OrganizationEventList(_, context), "events"),
    nextToken: import_smithy_client.expectString
  });
}, "de_DescribeEventsForOrganizationResponse");
var de_DescribeEventsResponse = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    events: /* @__PURE__ */ __name((_) => de_EventList(_, context), "events"),
    nextToken: import_smithy_client.expectString
  });
}, "de_DescribeEventsResponse");
var de_EntityList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_AffectedEntity(entry, context);
  });
  return retVal;
}, "de_EntityList");
var de_Event = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    availabilityZone: import_smithy_client.expectString,
    endTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "endTime"),
    eventScopeCode: import_smithy_client.expectString,
    eventTypeCategory: import_smithy_client.expectString,
    eventTypeCode: import_smithy_client.expectString,
    lastUpdatedTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "lastUpdatedTime"),
    region: import_smithy_client.expectString,
    service: import_smithy_client.expectString,
    startTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "startTime"),
    statusCode: import_smithy_client.expectString
  });
}, "de_Event");
var de_EventDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    event: /* @__PURE__ */ __name((_) => de_Event(_, context), "event"),
    eventDescription: import_smithy_client._json,
    eventMetadata: import_smithy_client._json
  });
}, "de_EventDetails");
var de_EventList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_Event(entry, context);
  });
  return retVal;
}, "de_EventList");
var de_OrganizationEvent = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    arn: import_smithy_client.expectString,
    endTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "endTime"),
    eventScopeCode: import_smithy_client.expectString,
    eventTypeCategory: import_smithy_client.expectString,
    eventTypeCode: import_smithy_client.expectString,
    lastUpdatedTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "lastUpdatedTime"),
    region: import_smithy_client.expectString,
    service: import_smithy_client.expectString,
    startTime: /* @__PURE__ */ __name((_) => (0, import_smithy_client.expectNonNull)((0, import_smithy_client.parseEpochTimestamp)((0, import_smithy_client.expectNumber)(_))), "startTime"),
    statusCode: import_smithy_client.expectString
  });
}, "de_OrganizationEvent");
var de_OrganizationEventDetails = /* @__PURE__ */ __name((output, context) => {
  return (0, import_smithy_client.take)(output, {
    awsAccountId: import_smithy_client.expectString,
    event: /* @__PURE__ */ __name((_) => de_Event(_, context), "event"),
    eventDescription: import_smithy_client._json,
    eventMetadata: import_smithy_client._json
  });
}, "de_OrganizationEventDetails");
var de_OrganizationEventList = /* @__PURE__ */ __name((output, context) => {
  const retVal = (output || []).filter((e) => e != null).map((entry) => {
    return de_OrganizationEvent(entry, context);
  });
  return retVal;
}, "de_OrganizationEventList");
var deserializeMetadata = /* @__PURE__ */ __name((output) => ({
  httpStatusCode: output.statusCode,
  requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
  extendedRequestId: output.headers["x-amz-id-2"],
  cfId: output.headers["x-amz-cf-id"]
}), "deserializeMetadata");
var throwDefaultError = (0, import_smithy_client.withBaseException)(HealthServiceException);
var buildHttpRpcRequest = /* @__PURE__ */ __name(async (context, headers, path, resolvedHostname, body) => {
  const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
  const contents = {
    protocol,
    hostname,
    port,
    method: "POST",
    path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
    headers
  };
  if (resolvedHostname !== void 0) {
    contents.hostname = resolvedHostname;
  }
  if (body !== void 0) {
    contents.body = body;
  }
  return new import_protocol_http.HttpRequest(contents);
}, "buildHttpRpcRequest");
function sharedHeaders(operation) {
  return {
    "content-type": "application/x-amz-json-1.1",
    "x-amz-target": `AWSHealth_20160804.${operation}`
  };
}
__name(sharedHeaders, "sharedHeaders");

// src/commands/DescribeAffectedAccountsForOrganizationCommand.ts
var DescribeAffectedAccountsForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeAffectedAccountsForOrganization", {}).n("HealthClient", "DescribeAffectedAccountsForOrganizationCommand").f(void 0, void 0).ser(se_DescribeAffectedAccountsForOrganizationCommand).de(de_DescribeAffectedAccountsForOrganizationCommand).build() {
  static {
    __name(this, "DescribeAffectedAccountsForOrganizationCommand");
  }
};

// src/commands/DescribeAffectedEntitiesCommand.ts



var DescribeAffectedEntitiesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeAffectedEntities", {}).n("HealthClient", "DescribeAffectedEntitiesCommand").f(void 0, void 0).ser(se_DescribeAffectedEntitiesCommand).de(de_DescribeAffectedEntitiesCommand).build() {
  static {
    __name(this, "DescribeAffectedEntitiesCommand");
  }
};

// src/commands/DescribeAffectedEntitiesForOrganizationCommand.ts



var DescribeAffectedEntitiesForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeAffectedEntitiesForOrganization", {}).n("HealthClient", "DescribeAffectedEntitiesForOrganizationCommand").f(void 0, void 0).ser(se_DescribeAffectedEntitiesForOrganizationCommand).de(de_DescribeAffectedEntitiesForOrganizationCommand).build() {
  static {
    __name(this, "DescribeAffectedEntitiesForOrganizationCommand");
  }
};

// src/commands/DescribeEntityAggregatesCommand.ts



var DescribeEntityAggregatesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEntityAggregates", {}).n("HealthClient", "DescribeEntityAggregatesCommand").f(void 0, void 0).ser(se_DescribeEntityAggregatesCommand).de(de_DescribeEntityAggregatesCommand).build() {
  static {
    __name(this, "DescribeEntityAggregatesCommand");
  }
};

// src/commands/DescribeEntityAggregatesForOrganizationCommand.ts



var DescribeEntityAggregatesForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEntityAggregatesForOrganization", {}).n("HealthClient", "DescribeEntityAggregatesForOrganizationCommand").f(void 0, void 0).ser(se_DescribeEntityAggregatesForOrganizationCommand).de(de_DescribeEntityAggregatesForOrganizationCommand).build() {
  static {
    __name(this, "DescribeEntityAggregatesForOrganizationCommand");
  }
};

// src/commands/DescribeEventAggregatesCommand.ts



var DescribeEventAggregatesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEventAggregates", {}).n("HealthClient", "DescribeEventAggregatesCommand").f(void 0, void 0).ser(se_DescribeEventAggregatesCommand).de(de_DescribeEventAggregatesCommand).build() {
  static {
    __name(this, "DescribeEventAggregatesCommand");
  }
};

// src/commands/DescribeEventDetailsCommand.ts



var DescribeEventDetailsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEventDetails", {}).n("HealthClient", "DescribeEventDetailsCommand").f(void 0, void 0).ser(se_DescribeEventDetailsCommand).de(de_DescribeEventDetailsCommand).build() {
  static {
    __name(this, "DescribeEventDetailsCommand");
  }
};

// src/commands/DescribeEventDetailsForOrganizationCommand.ts



var DescribeEventDetailsForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEventDetailsForOrganization", {}).n("HealthClient", "DescribeEventDetailsForOrganizationCommand").f(void 0, void 0).ser(se_DescribeEventDetailsForOrganizationCommand).de(de_DescribeEventDetailsForOrganizationCommand).build() {
  static {
    __name(this, "DescribeEventDetailsForOrganizationCommand");
  }
};

// src/commands/DescribeEventsCommand.ts



var DescribeEventsCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEvents", {}).n("HealthClient", "DescribeEventsCommand").f(void 0, void 0).ser(se_DescribeEventsCommand).de(de_DescribeEventsCommand).build() {
  static {
    __name(this, "DescribeEventsCommand");
  }
};

// src/commands/DescribeEventsForOrganizationCommand.ts



var DescribeEventsForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEventsForOrganization", {}).n("HealthClient", "DescribeEventsForOrganizationCommand").f(void 0, void 0).ser(se_DescribeEventsForOrganizationCommand).de(de_DescribeEventsForOrganizationCommand).build() {
  static {
    __name(this, "DescribeEventsForOrganizationCommand");
  }
};

// src/commands/DescribeEventTypesCommand.ts



var DescribeEventTypesCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeEventTypes", {}).n("HealthClient", "DescribeEventTypesCommand").f(void 0, void 0).ser(se_DescribeEventTypesCommand).de(de_DescribeEventTypesCommand).build() {
  static {
    __name(this, "DescribeEventTypesCommand");
  }
};

// src/commands/DescribeHealthServiceStatusForOrganizationCommand.ts



var DescribeHealthServiceStatusForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DescribeHealthServiceStatusForOrganization", {}).n("HealthClient", "DescribeHealthServiceStatusForOrganizationCommand").f(void 0, void 0).ser(se_DescribeHealthServiceStatusForOrganizationCommand).de(de_DescribeHealthServiceStatusForOrganizationCommand).build() {
  static {
    __name(this, "DescribeHealthServiceStatusForOrganizationCommand");
  }
};

// src/commands/DisableHealthServiceAccessForOrganizationCommand.ts



var DisableHealthServiceAccessForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "DisableHealthServiceAccessForOrganization", {}).n("HealthClient", "DisableHealthServiceAccessForOrganizationCommand").f(void 0, void 0).ser(se_DisableHealthServiceAccessForOrganizationCommand).de(de_DisableHealthServiceAccessForOrganizationCommand).build() {
  static {
    __name(this, "DisableHealthServiceAccessForOrganizationCommand");
  }
};

// src/commands/EnableHealthServiceAccessForOrganizationCommand.ts



var EnableHealthServiceAccessForOrganizationCommand = class extends import_smithy_client.Command.classBuilder().ep(commonParams).m(function(Command, cs, config, o) {
  return [
    (0, import_middleware_serde.getSerdePlugin)(config, this.serialize, this.deserialize),
    (0, import_middleware_endpoint.getEndpointPlugin)(config, Command.getEndpointParameterInstructions())
  ];
}).s("AWSHealth_20160804", "EnableHealthServiceAccessForOrganization", {}).n("HealthClient", "EnableHealthServiceAccessForOrganizationCommand").f(void 0, void 0).ser(se_EnableHealthServiceAccessForOrganizationCommand).de(de_EnableHealthServiceAccessForOrganizationCommand).build() {
  static {
    __name(this, "EnableHealthServiceAccessForOrganizationCommand");
  }
};

// src/Health.ts
var commands = {
  DescribeAffectedAccountsForOrganizationCommand,
  DescribeAffectedEntitiesCommand,
  DescribeAffectedEntitiesForOrganizationCommand,
  DescribeEntityAggregatesCommand,
  DescribeEntityAggregatesForOrganizationCommand,
  DescribeEventAggregatesCommand,
  DescribeEventDetailsCommand,
  DescribeEventDetailsForOrganizationCommand,
  DescribeEventsCommand,
  DescribeEventsForOrganizationCommand,
  DescribeEventTypesCommand,
  DescribeHealthServiceStatusForOrganizationCommand,
  DisableHealthServiceAccessForOrganizationCommand,
  EnableHealthServiceAccessForOrganizationCommand
};
var Health = class extends HealthClient {
  static {
    __name(this, "Health");
  }
};
(0, import_smithy_client.createAggregatedClient)(commands, Health);

// src/pagination/DescribeAffectedAccountsForOrganizationPaginator.ts

var paginateDescribeAffectedAccountsForOrganization = (0, import_core.createPaginator)(HealthClient, DescribeAffectedAccountsForOrganizationCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/DescribeAffectedEntitiesForOrganizationPaginator.ts

var paginateDescribeAffectedEntitiesForOrganization = (0, import_core.createPaginator)(HealthClient, DescribeAffectedEntitiesForOrganizationCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/DescribeAffectedEntitiesPaginator.ts

var paginateDescribeAffectedEntities = (0, import_core.createPaginator)(HealthClient, DescribeAffectedEntitiesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/DescribeEventAggregatesPaginator.ts

var paginateDescribeEventAggregates = (0, import_core.createPaginator)(HealthClient, DescribeEventAggregatesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/DescribeEventTypesPaginator.ts

var paginateDescribeEventTypes = (0, import_core.createPaginator)(HealthClient, DescribeEventTypesCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/DescribeEventsForOrganizationPaginator.ts

var paginateDescribeEventsForOrganization = (0, import_core.createPaginator)(HealthClient, DescribeEventsForOrganizationCommand, "nextToken", "nextToken", "maxResults");

// src/pagination/DescribeEventsPaginator.ts

var paginateDescribeEvents = (0, import_core.createPaginator)(HealthClient, DescribeEventsCommand, "nextToken", "nextToken", "maxResults");
// Annotate the CommonJS export names for ESM import in node:

0 && (module.exports = {
  HealthServiceException,
  __Client,
  HealthClient,
  Health,
  $Command,
  DescribeAffectedAccountsForOrganizationCommand,
  DescribeAffectedEntitiesCommand,
  DescribeAffectedEntitiesForOrganizationCommand,
  DescribeEntityAggregatesCommand,
  DescribeEntityAggregatesForOrganizationCommand,
  DescribeEventAggregatesCommand,
  DescribeEventDetailsCommand,
  DescribeEventDetailsForOrganizationCommand,
  DescribeEventTypesCommand,
  DescribeEventsCommand,
  DescribeEventsForOrganizationCommand,
  DescribeHealthServiceStatusForOrganizationCommand,
  DisableHealthServiceAccessForOrganizationCommand,
  EnableHealthServiceAccessForOrganizationCommand,
  paginateDescribeAffectedAccountsForOrganization,
  paginateDescribeAffectedEntitiesForOrganization,
  paginateDescribeAffectedEntities,
  paginateDescribeEventAggregates,
  paginateDescribeEventTypes,
  paginateDescribeEventsForOrganization,
  paginateDescribeEvents,
  EntityStatusCode,
  EventScopeCode,
  InvalidPaginationToken,
  UnsupportedLocale,
  EventAggregateField,
  EventStatusCode,
  EventTypeCategory,
  ConcurrentModificationException
});
